### ISSUE TITLE
Upgrading to go1.23.0 and failed in CI for goanalysis_metalinter

### ISSUE THREAD

**Issue by sysulq** [golangci/golangci-lint#4912]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

```txt
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package pprof: could not load export data: no export data for "github.com/grafana/pyroscope-go/godeltaprof/internal/pprof" 
ERRO Running error: can't run linter goanalysis_metalinter
buildir: failed to load package pprof: could not load export data: no export data for "github.com/grafana/pyroscope-go/godeltaprof/internal/pprof" 
task: Failed to run task "default": exit status 3
```

Here is the ci link for details.
- https://github.com/go-kod/kod/actions/runs/10398020216/job/28794683744#step:6:396

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Hello,

To be able to process your issue as quickly as possible,
can you edit your issue to add missing elements, fix the invalid elements, and complete the sections?

| Section                                                       | Status |
|---------------------------------------------------------------|--------|
| Welcome                                                       | 🟠 |
| Description of the problem                                    | 🟠 |
| Version of golangci-lint                                      | 🔴 |
| Configuration                                                 | 🔴 |
| Go environment                                                | 🔴 |
| Verbose output of running                                     | 🔴 |
| A minimal reproducible example or link to a public repository | 🔴 |
| Validation                                                    | 🔴 |
| Supporter                                                     | 🔴 |

Legend:
- 🟢: The section is complete and valid.
- 🟠: The section is incomplete.
- 🔴: The section is missing or invalid.

**Comment by sysulq** [comment]
Fixed after upgrading pyroscope-go to v1.1.2, close this issue

https://github.com/go-kod/kod/pull/228/commits/6c762f05c8d327c408ce962d4551d28631279e96
